keps/sig-node/3619-supplemental-groups-policy/README.md

```diff
@@ -0,0 +1,876 @@
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+The proposal affects to the CRI implementations (e.g., containerd, cri-o, gVisor, etc.)
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+- How to track the support status in CRI implementations of this proposal?
+  - This feature is mainly implemented inside each CRI implementation.
+- How to feature-gate this feature in CRI implementations?
+
+## Design Details
+
+A new field named `SupplementalGroupsPolicy` will be introduced to `PodSecurityContext`:
+
+```go
+type PodSecurityContext struct {
+  ...
+	// A list of groups applied to the first process run in each container. 
+	// supplementalGroupsPolicy can control how groups will be calculated.
+	// Note that this field cannot be set when spec.os.name is windows.
+	// +optional
+	SupplementalGroups []int64
+	// supplementalGroupsPolicy defines behavior of applying supplementalGroups 
+	// to the first process run in each container.
+	// Valid values are "RespectGroupsInImage" and "IgnoreGroupsInImage". 
+	// If note specified, "RespectGroupsInImage" is used.
+	// Note that this field cannot be set when spec.os.name is windows.
+	// +optional
+	SupplementalGroupsPolicy *PodSecurityGroupsPolicy 
+}
+
+type PodSecurityGroupsPolicy string
+const (
+	// SecurityGroupsPolicyRespectGroupsInImage indicates that 
+  // it respects group memberships defined in the container image 
+  // for the container's primary UID in supplementary groups of the 
+  // first container process.
+  // The resultant its supplementary groups are 
+  // the container's primary GID, the fsGroup (if specified), 
+  // supplementalGroups(if specified), and the groups defined in 
+  // the container image for the container's primary UID.
+	SecurityGroupsPolicyRespectGroupsInImage PodSecurityGroupsPolicy = "RespectGroupsInImage"
+
+	// SecurityGroupsPolicyIgnoreGroupsInImage indicates that 
+  // it ignores group memberships defined in the container image 
+  // for the container's primary UID in supplementary groups of the 
+  // first container process.
+  // The resultant its supplementary groups are 
+  // the container's primary GID, the fsGroup (if specified), 
+  // supplementalGroups(if specified).
+	SecurityGroupsPolicyIgnoreGroupsInImage PodSecurityGroupsPolicy = "IgnoreGroupsInImage"
+)
+```
+
+cri-spec (both on `v1` and `v1alpha2`) also needs to be updated similarly because of the change:
```

Comment:
https://github.com/kubernetes/enhancements/pull/3620#discussion_r1095158441

I feel pretty strongly now that we need to return the "actual results" information and publish it in the pod status.  I know it's an expansion of scope, but after analyzing it, it's just TOO confusing for us to not tell the user what ultimately happened.